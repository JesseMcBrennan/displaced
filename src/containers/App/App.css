.App {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.header {
  height: 2.3em;
}

.contact-section {
  display: flex;
  justify-content: center;
}