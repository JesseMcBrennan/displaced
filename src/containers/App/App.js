import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Route, withRouter } from 'react-router-dom';
import { NavBar } from '../NavBar/NavBar';
import MemberContainer from '../MemberContainer/MemberContainer';
import Search from '../Search/Search';
import ContactForm from '../ContactForm/ContactForm';
import ContactPdf from '../ContactPdf/ContactPdf';
import './App.css';


export class App extends Component {
  render() {
    return (
      <div className="App">
        <Route path="/" component={NavBar}/>
        <Route exact path="/" component={Search}/>
        <Route exact path="/MemberContainer" render={({ history }) => 
          <MemberContainer history={history}/>} 
        />
        <div className='contact-section'>
          <Route exact path="/ContactForm" render={({ history }) => 
            <ContactForm history={history}/>} 
          />
          <Route exact path="/ContactPdf" render={() => 
            <ContactPdf 
            />} />
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  members: state.searchResults
});

App.propTypes = {
  members: PropTypes.object
};


export default withRouter(connect(mapStateToProps)(App));




