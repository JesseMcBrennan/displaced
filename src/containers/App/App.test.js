import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps } from './App';

describe('App', () => {
  let wrapper;
  let mockHistory;

  beforeEach(() => {
    mockHistory = {push: jest.fn()};
    wrapper = shallow(<App history={mockHistory}/>);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  describe('mapStateToProps', () => {
    it('should return an object', () => {
      const mockResult = { members: {} };

      const expected = {};

      const mappedProps = mapStateToProps(mockResult);

      expect(mappedProps).toEqual(expected);

    });
  });
});



