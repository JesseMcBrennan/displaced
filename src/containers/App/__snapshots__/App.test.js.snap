// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          exact={true}
          path="/MemberContainer"
          render={[Function]}
/>,
        <div
          className="contact-section"
>
          <Route
                    exact={true}
                    path="/ContactForm"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/ContactPdf"
                    render={[Function]}
          />
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/MemberContainer",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/ContactForm"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/ContactPdf"
              render={[Function]}
/>,
          ],
          "className": "contact-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/ContactForm",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/ContactPdf",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            exact={true}
            path="/MemberContainer"
            render={[Function]}
/>,
          <div
            className="contact-section"
>
            <Route
                        exact={true}
                        path="/ContactForm"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/ContactPdf"
                        render={[Function]}
            />
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/MemberContainer",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/ContactForm"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/ContactPdf"
                render={[Function]}
/>,
            ],
            "className": "contact-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/ContactForm",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/ContactPdf",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
